js/app.js

```diff
@@ -131,392 +100,140 @@ app.controller('CheckoutController', function($scope, $interval, Order, $httpPar
-        stop_interval();
-    }
-    else if( "EXPIRED" == data.status ){
-        //Expired
-        update_altcoin_status('expired');
-        stop_interval();
+    //Copy bitcoin address to clipboard
+    $scope.blockonomics_address_click = function() {
+        copy_to_clipboard("bnomics-address-copy");
     }
 
-    function gotWrongAmount(){
-        if( ["UNDERPAY_RECEIVED", "UNDERPAY_CONFIRMED", "OVERPAY_RECEIVED", "OVERPAY_CONFIRMED"].indexOf(data.payment_status) > -1 ){
-            return true;
-        }
-        return false;
-    }
-    function isPaymentReceived(){
-        if( ["PAYMENT_RECEIVED", "PAYMENT_CONFIRMED"].indexOf(data.payment_status) > -1 ){
-            return true;
-        }
-        return false;
-    }
-    function gotFunds(){
-        if(gotWrongAmount() || isPaymentReceived()){
-            return true;
-        }
-        return false;
+    //Copy bitcoin amount to clipboard
+    $scope.blockonomics_amount_click = function() {
+        copy_to_clipboard("bnomics-amount-copy");
     }
-    function needsRefund(){
-        if( ["EXPIRED", "NEEDS_REFUND", "WAITING_FOR_DEPOSIT"].indexOf(data.status) > -1 && gotFunds() ){
-            return true;
-        }
-        return false;
-    }
-    function isRefundAddress(){
-        if( 'refund_address' in data ){
-            return true;
-        }
-        return false;
+    //Copy bitcoin address to clipboard
+    $scope.try_again_click = function() {
+        location.reload();
     }
 }
 
-    //Add a refund address to altcoin order
-    $scope.add_refund_click = function() {
-        $scope.altrefund = document.getElementById("bnomics-refund-input").value;
-        $scope.altuuid = get_uuid();
-        var response = AltcoinAddRefund.save({
-            'uuid': $scope.altuuid,
-            'address': $scope.altrefund
-        },function successCallback(data) {
-            if(data.result == 'ok') {
-                send_refund_email();
-                update_altcoin_status('refunded');
-            }else if(data.errors){
-                var refund_message = document.getElementById("bnomics-refund-errors");
-                refund_message.innerHTML = "";
-                for (var key in data.errors){
-                    var value = data.errors[key];
-                    for (var i = value.length - 1; i >= 0; i--) {
-                        refund_message.innerHTML += "<p style='color:red'>"+value[i]+"</p>";
-                    }
-                } 
-            }
-        });
-    }
-
-    //Display more info about payment status, refund etc.
-    $scope.already_paid = function() {
-        $scope.show_refund_info = true;
-    }
-
-    //Go to add refund page
-    $scope.get_refund = function() {
-        stop_interval();
-        $scope.hide_refund_reason = true;
-        update_altcoin_status('add_refund');
-    }
+Order.$inject = ['$resource'];
+function Order($resource) {
+    //There are two styles of callback url in 
```

Comment:
This should also use the new suggested function *get_wc_endpoint*. Move these code comments inside *get_wc_endpoint* so that we better understand the need of that function

Reply:
Refactored in 130cef6